File: Tps_individuales/murcia_Tobias_Llop.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

string solve(const vector<int>& h, const vector<int>& w) {
    int n = h.size();
    vector<int> max_inc(n, 0);
    vector<int> max_dec(n, 0);

    // Calculamos los máximos incrementos
    for (int i = 0; i < n; ++i) {
        max_inc[i] = w[i]; // Inicializamos con el ancho del edificio actual
        for (int j = 0; j < i; ++j) {
            if (h[i] > h[j]) {
                max_inc[i] = max(max_inc[i], max_inc[j] + w[i]); // Actualizamos si encontramos un incremento mayor
            }
        }
    }

    // Calculamos los máximos decrementos
    for (int i = n - 1; i >= 0; --i) {
        max_dec[i] = w[i]; // Inicializamos con el ancho del edificio actual
        for (int j = n - 1; j > i; --j) {
            if (h[i] > h[j]) {
                max_dec[i] = max(max_dec[i], max_dec[j] + w[i]); // Actualizamos si encontramos un decremento mayor
            }
        }
    }

    int max_inc_global = *max_element(max_inc.begin(), max_inc.end()); // Encontramos el máximo global de incremento
    int max_dec_global = *max_element(max_dec.begin(), max_dec.end()); // Encontramos el máximo global de decremento

    // Determinamos cuál es mayor
    if (max_inc_global >= max_dec_global) {
        return "Increasing (" + to_string(max_inc_global) + "). Decreasing (" + to_string(max_dec_global) + ").";
    } else {
        return "Decreasing (" + to_string(max_dec_global) + "). Increasing (" + to_string(max_inc_global) + ").";
    }
}

int main() {
    int casos;
    cin >> casos;

    for (int caso = 1; caso <= casos; ++caso) {
        int cant_edificios;
        cin >> cant_edificios;
        vector<int> altura(cant_edificios);
        vector<int> ancho(cant_edificios);

        for (int i = 0; i < cant_edificios; ++i) {
            cin >> altura[i];
        }

        for (int i = 0; i < cant_edificios; ++i) {
            cin >> ancho[i];
        }

        string resultado = solve(altura, ancho);
        cout << "Case " << caso << ". " << resultado << endl;
    }

    return 0;
}

File: Tps_individuales/murcia_Tobias_Llop.py
"""
Entregable 2 para la materia AyEd III - LCD UBA
Tobias Llop
Link al enunciado: https://vjudge.net/problem/UVA-11790
"""

def max_creciente(h, w, n):
    dp = [0] * n
    for i in range(n):
        dp[i] = w[i]
        for j in range(i):
            if h[i] > h[j]:
                dp[i] = max(dp[i], dp[j] + w[i])
    return max(dp)

def max_decrec(h, w, n):
    dp = [0] * n
    for i in range(n):
        dp[i] = w[i]
        for j in range(i):
            if h[i] < h[j]:
                dp[i] = max(dp[i], dp[j] + w[i])
    return max(dp)

def solve(h, w, n):
    inc_dp = [w[i] for i in range(n)]  # Lista para el caso de aumento
    dec_dp = [w[i] for i in range(n)]  # Lista para el caso de disminución

    max_inc = w[0]  # Inicializar con el primer edificio para el caso de aumento
    max_dec = w[0]  # Inicializar con el primer edificio para el caso de disminución

    for i in range(1, n):
        for j in range(i):
            if h[i] > h[j]:
                inc_dp[i] = max(inc_dp[i], inc_dp[j] + w[i])
                max_inc = max(max_inc, inc_dp[i])
            elif h[i] < h[j]:
                dec_dp[i] = max(dec_dp[i], dec_dp[j] + w[i])
                max_dec = max(max_dec, dec_dp[i])

    # Determinar cuál es mayor
    if max_inc >= max_dec:
        return f"Increasing ({max_inc}). Decreasing ({max_dec})."
    else:
        return f"Decreasing ({max_dec}). Increasing ({max_inc})."

def main():
    casos = int(input())
    output = [None] * casos

    for caso in range(casos):
        cant_edificios = int(input())
        altura = list(map(int, input().split()))
        ancho = list(map(int, input().split()))
        output[caso] = solve(altura, ancho, cant_edificios)

    for caso, resultado in enumerate(output, start=1):
        print(f"Case {caso}. {resultado}")

if __name__ == "__main__":
    main()
